python/cuml/dask/ensemble/base.py

```diff
@@ -0,0 +1,180 @@
+                n_estimators=self.n_estimators_per_worker[n],
+                seed=seeds[n],
+                **kwargs,
+                pure=False,
+                workers=[worker],
+            )
+            for n, worker in enumerate(self.workers)
+        }
+
+        wait_and_raise_from_futures(list(self.rfs.values()))
+
+    def _estimators_per_worker(self, n_estimators):
+        n_workers = len(self.workers)
+        if n_estimators < n_workers:
+            raise ValueError(
+                "n_estimators cannot be lower than number of dask workers."
+            )
+
+        n_est_per_worker = math.floor(n_estimators / n_workers)
+        n_estimators_per_worker = \
+            [n_est_per_worker for i in range(n_workers)]
+        remaining_est = n_estimators - (n_est_per_worker * n_workers)
+        for i in range(remaining_est):
+            n_estimators_per_worker[i] = (
+                n_estimators_per_worker[i] + 1
+            )
+        return n_estimators_per_worker
+
+    def _fit(self, model, dataset, convert_dtype):
+        data = DistributedDataHandler.create(dataset, client=self.client)
+        self.datatype = data.datatype
+        futures = list()
+        for idx, (worker, worker_data) in \
+                enumerate(data.worker_to_parts.items()):
+            futures.append(
+                self.client.submit(
+                    _func_fit,
+                    model[worker],
+                    worker_data,
+                    convert_dtype,
+                    workers=[worker],
+                    pure=False)
+            )
+        wait_and_raise_from_futures(futures)
+        return self
+
+    def _concat_treelite_models(self):
+        """
+        Convert the cuML Random Forest model present in different workers to
+        the treelite format and then concatenate the different treelite models
+        to create a single model. The concatenated model is then converted to
+        bytes format.
+        """
+        model_protobuf_futures = list()
+        for w in self.workers:
+            model_protobuf_futures.append(
+                dask.delayed(_get_protobuf_bytes)
+                (self.rfs[w]))
+        mod_bytes = self.client.compute(model_protobuf_futures, sync=True)
+        last_worker = w
+        all_tl_mod_handles = []
+        model = self.rfs[last_worker].result()
+        all_tl_mod_handles = [model._tl_model_handles(pbuf_bytes)
+                              for pbuf_bytes in mod_bytes]
+
+        model._concatenate_treelite_handle(
+            treelite_handle=all_tl_mod_handles)
+        return model
+
+    def _predict_using_fil(self, X, delayed, **kwargs):
+        data = DistributedDataHandler.create(X, client=self.client)
+        self.datatype = data.datatype
+        return self._predict(X, delayed=delayed, **kwargs)
+
+    def _get_params(self, deep):
+        model_params = list()
+        for idx, worker in enumerate(self.workers):
+            model_params.append(

```

Comment:
Future PR - no need to address now... This pattern seems to recur a lot (at least 3 places in RF maybe more in other modes). Should there be a helper "results = map_over_rfs(function, **params)" that does this loop over all rf/worker pairs, runs "function" on each worker (with self.rfs[worker] as first arg) and returns the results as a list or dict?